### ISSUE TITLE
Need ability to get component pathing from driver URL

### ISSUE THREAD

**Issue by leeviana** [microsoft/FluidFramework#4490]
Today we have a custom hardcoded path for ODSP drivers to get component pathing from a driver URL (the opposite functionality of getAbsoluteUrl). We want to replace this with Framework native support. Either an interface on all resolvers that fulfills this contract ("resolveDataStore") or some other helper similar to what Loader.request() does to request the component path.

Related to the discussion here: https://github.com/microsoft/FluidFramework/issues/4169 which concluded we can use Loader.request() instead of needing this functionality. However, we are blocked on using Loader.request() because of container caching issues: https://github.com/microsoft/FluidFramework/issues/4489

@agarwal-navin @anthony-murphy @jatgarg 

This is related to a core Fluid Framework scenario: Going from a Fluid resource URL to a Fluid object using the Loader object. Today without proper support for Loader.request (see #4489 and #4498), the only way for us to load a component is to parse out the component path from the URL separately using a SPO driver custom export and then call Loader.resolve() and Container.request(). This seems like one of the first promises that the Fluid Framework makes - that you can directly use the Loader object to load a Fluid object and should likely be fixed. :)

**Comment by tylerbutler** [comment]
@leeviana Can you create a new issue describing the scenario that you're trying to address, and link these issues to the new uber-issue? That wasx we can better understand what you're doing and what changes are needed in the framework.

**Comment by leeviana** [comment]
Yeah, I can do that, this one is the most fundamental of Framework capabilities, going from a URL to a Fluid component using the Loader. :P Do we need another issue or should I just edit this one to be more scenario oriented? Otherwise I don't think there's a need for this issue any longer if I create a duplicate.

**Comment by leeviana** [comment]
Updated the issue with information about needing to go from a Fluid URL to a Fluid object using the Loader object.

**Comment by tylerbutler** [comment]
@leeviana

> Related to the discussion here: #4169 which concluded we can use Loader.request() instead of needing this functionality. However, we are blocked on using Loader.request() because of container caching issues: #4489

I'm a bit confused. Per #4169 the ultimate solution seems to be using Loader.request, which you're blocked on from #4489. This leads me to think that the issue here is a _temporary_ solution? The fix for #4489 is scheduled for January; I'm not inclined to do anything temporary when a real fix is coming soon.

**Comment by leeviana** [comment]
Is it confirmed that #4489 is happening in the January timeframe? If it is, then of course we should wait on it, but all I've seen so far is general discussion about what work could happen if someone wanted to do it and the January tag which doesn't mean much since issues just get rolled over from month to month. If it is happening sometime soonish then hopefully we won't need this, but this is a pretty core piece of Fluid functionality to leave broken for so long.

**Comment by curtisman** [comment]
I am closing this as #4489 is the right solution.    @heliocliu for visibility about this dependency.

**Comment by curtisman** [comment]
@anthony-murphy There seems to be quite a bit of confusion the over what is actually needed.  From reading all these thread.  It seems that #4489 should be sufficient to unblock @leeviana.   Is there addition capability that is fundamental in this area that is missing?

**Comment by leeviana** [comment]
Yes, there might be other needs for generic component path parsing, but not ones that the core loader flows need right now. #4489 (and #4491) will directly enable us to get rid of componentPath logic entirely in our loading pipeline as intended by the Framework.

**Comment by leeviana** [comment]
Since both #4489 and #4491 are closed, let's close this one for now. I think this is probably still an API reuse/need someone will need but for our specific loading scenario we avoid this problem with the resolution of those two bugs.
